### PR TITLE
Emotion-ferplus-8 implemented

### LINKED ISSUE #8: use a smaller model
Download and save pretrained  ONNX model
Verify model loads successfully with `onnxruntime`

### PR DESCRIPTION
**Pull request by spyke7**
hi @NevroHelios and @sayanChaterjee 
I have implemented the [emotion-ferplus-8 model](https://github.com/onnx/models/tree/main/validated/vision/body_analysis/emotion_ferplus), with opset version 8.
This resolves #8 

- used haarcascade from cv2 to detect face from the PIL image
- added a minimum padding to the gray scale image
- used scoremax function to generate probabilites of different emotion class

In the demo page I have added a option menu showing, the previously implemented ViT model from hugging face as well the new model. Also I have shown probabilites percentage. 

Also, there is commented out code for changing the opset version of the model.

Please review this

took reference from - https://bleedaiacademy.com/facial-expression-recognition-emotion-recognition-with-opencv/ 

## Summary by CodeRabbit

* **New Features**
  * Demo adds a model selector (two model options) and displays per-emotion confidence probabilities with predictions.
  * Backend API now accepts a model selection and returns emotion plus per-class probabilities.

* **Documentation**
  * API docs and examples updated to show model option usage and responses including probabilities; notes for the alternative model added.

* **Chores**
  * .gitignore updated to exclude Python cache and local env dirs; requirements updated to include ONNX support.

### CODE REVIEW

File: model-lab/README.md [hunk 1]
```diff
@@ -43,10 +43,13 @@ The API will start at:
...
-
+Give n = 1 or 2
+- 1 - For using the ViT model from hugging face
+- 2 - For using the Emotion-Ferplus-8 model as mentioned in onnx
 ```bash
 curl -X POST "http://127.0.0.1:8000/predict" \
-  -F "file=@face.jpg"
+  -F "file=@face.jpg" \
+  -F "model_option={n}"
 ```
```
[the remark this answers is not in the document]
Comment (spyke7) [reply model-lab/README.md:53]: Check now

File: model-lab/app/model.py [hunk 2]
```diff
@@ -1,16 +1,96 @@
...
+            probablities =  expanded / expanded.sum()
+
+            prob = np.squeeze(probablities)
+
+            for emotion, idx in sorted(self.emotion_table.items(), key=lambda x: x[1]):
+                self.prob_precentages[emotion] = float(prob[idx]*100)
+                
+            key = [k for k, v in self.emotion_table.items() if v == prob.argmax()][0]
+            label = key
+            return label, self.prob_precentages
```
[the remark this answers is not in the document]
Comment (spyke7) [reply model-lab/app/model.py]: check now

File: model-lab/app/model.py [hunk 3]
```diff
@@ -1,16 +1,98 @@
...
-        logits = model(**inputs).logits
-        predicted = logits.argmax(-1).item()
-    label = model.config.id2label[predicted]
-    return label
+            self.net=cv2.dnn.readNetFromONNX(model_path)
+    
+    def detect_face(self, pil_image):
+        image = cv2.cvtColor(np.array(pil_image), cv2.COLOR_RGB2BGR)
+        gray = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
+        faces = self.face_cascade.detectMultiScale(
```
Comment (NevroHelios) [inline model-lab/app/model.py:35]: no need for robust preprocessing before inference, rather we will move towards finetuning the model for different envs

File: model-lab/app/model.py [hunk 4]
```diff
@@ -1,16 +1,98 @@
...
 from PIL import Image
+import numpy as np
+import onnxruntime as ort
+import onnx
+from onnx import version_converter
+import cv2
 import torch
 
-MODEL_NAME = "trpakov/vit-face-expression"
+class Model:
```
Comment (NevroHelios) [inline model-lab/app/model.py:11]: use a config.py and there specify the table and model name and other constants

File: model-lab/app/model.py (hunk 3 above)
Comment (spyke7) [reply model-lab/app/model.py:35]: Can you please do that. Actually I'm still learning about that.. 😅

File: model-lab/app/model.py (hunk 4 above)
Comment (spyke7) [reply model-lab/app/model.py:11]: Done

File: model-lab/README.md [hunk 5]
```diff
@@ -43,14 +43,45 @@ The API will start at:
...
+      "neutral":0.6067721843719482,
+      "happiness":0.128993958234787,
+      "surprise":0.22512483596801758,
+      "sadness":0.025787757709622383,
+      "anger":96.17911529541016,
+      "disgust":2.6538150310516357,
+      "fear":0.04158278927206993,
+      "contempt":0.13880738615989685
+    }
+}
```
[the remark this answers is not in the document]
Comment (spyke7) [reply model-lab/README.md:69]: no they sum upto around 100

**Comment by spyke7** [comment]
@NevroHelios 
I cannot fine tune the model now but will surely work on that. Please review other things as well.
@sayanChaterjee please review it

File: model-lab/app/model.py (hunk 3 above)
Comment (NevroHelios) [reply model-lab/app/model.py:35]: no need to add finetuning rn, just remove the preprocessing and do prediction directly

Comment (spyke7) [reply model-lab/app/model.py:35]: in the preprocess function, just the pil image is just resized and then to np float32 type. I have done this as this was mentioned here - https://github.com/onnx/models/tree/main/validated/vision/body_analysis/emotion_ferplus
For the prediction, the image should be as np float32 type and reshaped to the input_shape as mentioned.
For the detect_face function, that can be omitted but then the accuracy will be less.

**Review by sayanChaterjee — APPROVED** [review]

**Review by NevroHelios — APPROVED** [review]
lgtm
